integration/token/fungible/views/accept.go

```diff
@@ -72,5 +71,16 @@ func (a *AcceptCashView) Call(context view.Context) (interface{}, error) {
 	assert.NoError(err, "failed to retrieve vault status for transaction [%s]", tx.ID())
 	assert.Equal(network.Valid, vc, "transaction [%s] should be in valid state", tx.ID())
 
+	// Check that the tokens are in the vault
+	qe := vault.TokenVault().QueryEngine()
+	for _, output := range outputs.ByRecipient(id).Outputs() {
+		tokenID := &token2.ID{TxId: tx.ID(), Index: output.Index}
+		_, toks, err := qe.GetTokens(tokenID)
```

Comment:
It might be more efficient to aggregate the token identifiers and then do a query for all of them.

Reply:
this is just an example. I think it is okay, no?